src/main/java/org/commcare/suite/model/MenuLoader.java

```diff
@@ -145,6 +145,14 @@ private boolean menuIsRelevant(SessionWrapperInterface sessionWrapper, Menu m) t
         return true;
     }
 
+    public boolean menuAssertionsPass(SessionWrapperInterface sessionWrapper, Menu m) {
+        Text text =  m.getAssertions().getAssertionFailure(sessionWrapper.getEvaluationContext());
+        if (text != null) {
+            loadException = new Exception(text.evaluate());
+        }
+        return true;
```

Comment:
shouldn't this return false if the assertion failed?

Reply:
fixed here https://github.com/dimagi/commcare-core/pull/1266/commits/61fb618032bba12b414049e2b0bbefdf7ba44296